task8/TutterApp/src/main/java/dao/PostDAOImpl.java

```diff
@@ -0,0 +1,112 @@
+package dao;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import forms.FilterForm;
+import models.Constants;
+import models.Post;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class PostDAOImpl implements PostDAO {
+    private static List<Post> posts;
+
+    public PostDAOImpl() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.setDateFormat(new SimpleDateFormat(Constants.DATE_FORMAT));
+
+        try {
+            posts = objectMapper.readValue(new File(Constants.PATH), new TypeReference<>() {
+            });
+        } catch (IOException e) {
+            posts = new ArrayList<>();
+
+            System.out.println(e.getMessage());
```

Comment:
OK for now but please avoid logging to stdout in server side development. This is useless, you see: if you are running a project and have problems, you cannot look for debug messages anywhere unless you sit and  stare directly in the console (but imagine problems started e.g. overnight!).
Error messages **must** be logged to a file or a DB. Therefore, dedicated logger is an absolute must have. Use `slf4j` or whatever